spec/controllers/coronavirus_form/check_answers_controller_spec.rb

```diff
@@ -28,13 +29,45 @@
 
     it "saves the form response to the database" do
       session["attribute"] = "key"
+      session["medical_equipment"] = I18n.t("coronavirus_form.questions.medical_equipment.options.option_yes.label")
+      session["contact_details"] = { "contact_name" => "Harry Potter" }
       post :submit
 
       expect(FormResponse.first.form_response).to eq(
-        [%w[attribute key], %w[reference_number abc]],
+        "attribute" => "key",
+        "contact_details" => { "contact_name" => "Harry Potter" },
+        "medical_equipment" => "Yes",
+        "reference_number" => "abc",
       )
     end
 
+    context "when unwanted fields are present in the session" do
+      it "excludes those fields from the FormResponse" do
+        data = valid_data
```

Comment:
This assignment isn't really need. `valid_data` could be used instead.

Reply:
`valid_data` is not deterministic, since it uses `SecureRandom.uuid` and `[].sample`, so I use the assignment so that I can use the same `data` in the expectation and the form submission.

Reply:
I forgot about the SecureRandoms. I wonder who put them there? 🤔 😊